src/js/components/help/downloads.js

```diff
@@ -51,7 +51,9 @@ const defaultLocationFormatter = ({ os, tool, versionInfo }) => {
 const osArchLocationReducer = ({ archList, location = downloadLocations.public, packageId, id, osList, versionInfo }) =>
   osList.reduce((accu, os) => {
     const osArchitectureLocations = archList.map(arch => ({
-      location: `${location}/repos/debian/pool/main/${id[0]}/${id}/${encodeURIComponent(`${packageId}_${getVersion(versionInfo, id)}-1+${os}_${arch}.deb`)}`,
+      location: `${location}/repos/debian/pool/main/${id[0]}/${packageId || id}/${encodeURIComponent(
+        `${packageId}_${getVersion(versionInfo, id)}-1+${os}_${arch}.deb`
+      )}`,
```

Comment:
Sorry, this is a bit more tricky than anticipated (after all, we are reverse-engineering the APT repositories pool). Two things still missing:
* one "recipe" can have multiple "packages". For example `mender-client` recipe produces `mender-client` and `mender-client-dev`, but both end up in the same recipe directory. So basically URL for the `-dev` one will be: `https://downloads.mender.io/repos/debian/pool/main/m/mender-client/mender-client-dev_3.3.0...`
* on top of that, some packages are architecture agnostic and need to have `_all` suffix. This applies to `mender-client-dev`: `.../main/m/mender-client/mender-client-dev_3.3.0-1%2Bdebian%2Bbuster_all.deb`

Reply:
the `all` suffix is already used for monitor & configure, but I was assuming the architecture would be the same for all the packages... would you think it's only limited to the `-dev` package that this diverges to `all` or would you expect this to happen more often?

Reply:
As a rule of thumb we should not assume that all packages from one recipe share architecture dependencies. The `-dev` here is a coincidence: `mender-client`(recipe) produces a package with binaries (iow architecture dependent) and another with dev files (architecture independent). But instead of `-dev` could be an extra package with examples, or documentation, or something else that is architecture independent.

In mender-configure we have example packages that are architecture independent - in this case the main package is also architecture independent but it could very well not be the case.